homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -39,19 +43,24 @@ class KeeneticFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     @staticmethod
     @callback
-    def async_get_options_flow(config_entry):
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> KeeneticOptionsFlowHandler:
         """Get the options flow for this handler."""
         return KeeneticOptionsFlowHandler(config_entry)
 
-    async def async_step_user(self, user_input=None):
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> dict[str, Any]:
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            host = self.context.get(CONF_HOST, user_input.get(CONF_HOST))
```

Comment:
```suggestion
            host = self.context.get(CONF_HOST, user_input[CONF_HOST])
```

Reply:
The suggestion would lead to KeyError since `user_input[CONF_HOST]` is evaluated before and irrespective of context getting result.

But I see your intention. The final form I use now is as follows:
`self.context.get(CONF_HOST) or user_input[CONF_HOST]`